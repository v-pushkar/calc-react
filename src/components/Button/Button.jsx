import React from "react";
import classNames from "classnames";

export const Button = (props) => {
  const { label, onClick, className, value, type, disabled } = props;
  const classes = classNames("btn", { [className]: className });

  return (
    <div className="btn-wrap" data-testid="button">
      <button
        className={classes}
        onClick={onClick}
        value={value}
        data-type={type}
        disabled={disabled}
      >
        {label}
      </button>
    </div>
  );
};
