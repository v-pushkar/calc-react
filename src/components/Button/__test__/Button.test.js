import React from "react";
import ReactDOM from "react-dom";
import {Button} from "./../Button";
import {render, cleanup} from "@testing-library/react";
import "@testing-library/jest-dom/extend-expect";
import renderer from "react-test-renderer"

afterEach(cleanup)
it("renders button without crashing",()=>{
    const div = document.createElement("div");
    ReactDOM.render(<Button/>,div)
})

it("renders button correctly 'value' ",()=>{
   const {getByTestId}= render(<Button  
        
        label={"value"}
    ></Button>)
    expect(getByTestId("button").textContent).toBe("value")
})
it("renders button correctly 9",()=>{
   const {getByTestId}= render(<Button 
        
        label={"9"}
    ></Button>)
    expect(getByTestId("button").textContent).toBe("9")
})

it("mutches snapshot button",()=>{
    const tree = renderer.create(<Button label="8"></Button>)
    expect(tree).toMatchSnapshot()
})
it("mutches snapshot button 2",()=>{
    const tree = renderer.create(<Button label="+"></Button>)
    expect(tree).toMatchSnapshot()
})