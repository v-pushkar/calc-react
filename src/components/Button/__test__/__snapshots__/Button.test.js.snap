// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`mutches snapshot button 1`] = `
<div
  className="btn-wrap"
  data-testid="button"
>
  <button
    className="btn"
  >
    8
  </button>
</div>
`;

exports[`mutches snapshot button 2 1`] = `
<div
  className="btn-wrap"
  data-testid="button"
>
  <button
    className="btn"
  >
    +
  </button>
</div>
`;
